modules/core/client/components/README.md

```diff
@@ -1,5 +1,36 @@
 # Reusable components
 
+## Navigation
+
+A `Back`, `Next` and `Finish` button. Both large and small screens. As seen in `/offer/host` and `/profile/{username}/references/new`.
+
+You need to provide state from outside.
+
+### props
+
+- `errors` - array of arrays of error messages for each tab
+  When there are errors for a tab, the `Next` button will be disabled with tooltip displaying the first of the errors.
+
+### Usage
+
+```jsx
+import Navigation from `@/modules/core/client/components/Navigation`;
+
+// ...
+<Navigation
+  tab={currentTabNumber}
+  tabs={totalTabNumber}
+  errors={arrayOfArraysOfErrorsForEachTab}
```

Comment:
Seems little confusing. Can we simplify this?

Reply:
> Seems little confusing. Can we simplify this?

@simison, do you mean simplify the variable name or API?

On side note: This is a typical scenario where I :heart: Typescript. Not that I'm ready to start migrating Trustroots to Typescript any time soon, or ever. :laughing:

Reply:
API. :-) Not a blocker, just I felt like doing some brain-gymnastics when trying to grok how to use this component and "Ask @mrkvon if you need explanation." comment was a flag towards some complexity too. 😁 

Not saying these are better but just wanted to share for inspiration:
- https://wpcalypso.wordpress.com/devdocs/design/wizard
- https://wpcalypso.wordpress.com/devdocs/design/wizard-progress-bar

Reply:
> API

Fair enough. Thank you for the examples! I get some inspiration how to at least name the props more clearly...

> Ask @mrkvon if you need explanation.

Lol that really is a flag... :grin: If one is too lazy or tired to write the API docs, one should consider changing the API...

Currently I'm thinking about:

- `currentStep` instead of `tab`
- `numberOfSteps` instead of `tabs`
- leave `onBack`, `onNext`, `onSubmit` as it is

I still don't know what to do about `disabled` and `errors`. They're both meant to block the `Next` or `Finish` buttons.
- `errors` - when `numberOfSteps={3}`, `[[], ['error', 'error2'], []]` would disable progress from second and third tab, and show text `error` in the tooltip above the disabled button. Meant primarily for form validation.
- `disabled` serves to disable `Finish` button e.g. when a form gets saved to API. Alternatively it could be replaced with `errors=[[], [], ['saving']]`, so it could be completely removed.

I can imagine removing the `disabled` and renaming the `errors` to something like `progressErrors`, but I'm not sure about a simpler data structure.
Well, now that I think about it, maybe we can pass just the single progress error applicable to the current step.
e.g. instead of both `errors` and `disabled` we can have only one of the following:
```js
progressError: PropTypes.string // not required
```
or
```js
disabled: PropTypes.oneOf(PropTypes.string, PropTypes.boolean) // not required
```


Do you have ideas about the above or how to further simplify the API; or change it completely?

Reply:
Would using `disabled` props (for all and/or each button separately) simplify it? That way consumer could just `disabledNext={ isError }` etc.

Reply:
> Would using disabled props (for all and/or each button separately) simplify it?

Yes, I think so. It sounds like a reasonable direction to simplify this.
Just `disabled` would do it, since we only disable forward buttons, ever.
However we need to pass the message for the tooltip, too. So I wonder whether to name it `disabled` (which i'd expect to be boolean, not string) or something arbitrary like `progressError`. Or pass both.
WDYT?

Reply:
I've started to think we should use less tooltips, since they're hidden info and if the message is important, it should just outright be visible.

`disabledReason` seems like good optional prop otherwise?

Reply:
@simison That makes sense. I'll be off the screen now and will implement later. Thank you for the guidance.

Reply:
Changed in 711113df9d86af557ddb5dd164d9777f2370ff1e